### PR TITLE
Product property persistence working

### PR DESCRIPTION
**Pull request by wcollins37**

### CODE REVIEW

File: nile-backend/src/main/java/com/tp/Nile/models/Product.java [hunk 1]
```diff
@@ -27,15 +27,15 @@
     private Integer productId;
 
     @ManyToOne(fetch = FetchType.EAGER,
-            cascade = { CascadeType.PERSIST })
+            cascade = { CascadeType.MERGE })
```
Comment (zmiles17) [inline nile-backend/src/main/java/com/tp/Nile/models/Product.java:30]: I was saying to add the merge property along with the persist property. Not delete it.

File: nile-backend/src/main/java/com/tp/Nile/services/ProductServiceImpl.java [hunk 2]
```diff
@@ -65,7 +68,58 @@ public Product getProductById(Integer productId) throws InvalidProductIdExceptio
 
     }
 
-    public Product addProduct(Product product) {
+    public Product addProduct(Product product) throws MissingPropertyException {
```
Comment (zmiles17) [inline nile-backend/src/main/java/com/tp/Nile/services/ProductServiceImpl.java:71]: Are you sure this logic is necessary? Did you write any tests before writing this?

**Review by zmiles17 — CHANGES_REQUESTED** [review]

File: nile-backend/src/main/java/com/tp/Nile/models/Product.java (hunk 1 above)
Comment (wcollins37) [reply nile-backend/src/main/java/com/tp/Nile/models/Product.java:30]: Just tried that, still getting detached entity error in Postman

**Review by Jason-Williams-github — COMMENTED** [review]
everything else looks good other than what Zach suggested.

**Review by tamera-brown — APPROVED** [review]

File: nile-backend/src/main/java/com/tp/Nile/services/ProductServiceImpl.java (hunk 2 above)
Comment (wcollins37) [reply nile-backend/src/main/java/com/tp/Nile/services/ProductServiceImpl.java:71]: When just changing the cascade types and not adding additional logic, the same error occurs

**Pull request closed without merging**